ui-web/sdk/src/TrinsicPopup.ts

```diff
@@ -0,0 +1,356 @@
+export class TrinsicPopup {
+    protected readonly popup: Window;
+    protected launchUrl?: string;
+    protected sessionId?: string;
+
+    private _messageListener?: (event: MessageEvent) => void;
+    private _broadcastChannel?: BroadcastChannel;
+    private _hasInitialized: boolean = false;
+    private _hasDisposed: boolean = false;
+    private _hasReceivedFinalizationSignal: boolean = false;
+    private _pollingIndicatedSuccess: boolean = false;
+
+    constructor(popup: Window) {
+        this.popup = popup;
+    }
+
+    /**
+     * Whether the popup window is definitely still open.
+     * 
+     * This may return false negatives (i.e., this function may return "false" even if the window is still open).
+     * This is because the `Cross-Origin-Opener-Policy` security header, if set on any page a popup visits (e.g. an Identity Provider's page), will
+     * cause the browser to sever the connection an opening window has to the popup, making it impossible to check if it's closed.
+     * 
+     * If this functions returns `true`, the popup window is open.
+     * If this function returns `false`, the popup window MAY be closed. Offer the user the opportunity to cancel out of the Session, but do not automatically do so.
+     */
+    get isDefinitelyOpen() {
```

Comment:
I think if there is a possibility for it to return a false negative, it shouldn't be called definitely.